plugin-tray/trayicon.cpp

```diff
@@ -196,6 +196,9 @@ void TrayIcon::init()
     XMapWindow(dsp, mIconId);
     XMapRaised(dsp, mWindowId);
 
+    QRect rectangle = iconGeometry();
```

Comment:
Hm... isn't it enough to create the window correctly positioned? Please, consider this diff:
```diff
diff --git a/plugin-tray/trayicon.cpp b/plugin-tray/trayicon.cpp
index 43cbb5d2..6760016e 100644
--- a/plugin-tray/trayicon.cpp
+++ b/plugin-tray/trayicon.cpp
@@ -126,7 +126,8 @@ void TrayIcon::init()
     set_attr.border_pixel = 0;
     mask = CWColormap|CWBackPixel|CWBorderPixel;
 
-    mWindowId = XCreateWindow(dsp, this->winId(), 0, 0, mIconSize.width() * metric(PdmDevicePixelRatio), mIconSize.height() * metric(PdmDevicePixelRatio),
+    const QRect icon_geom = iconGeometry();
+    mWindowId = XCreateWindow(dsp, this->winId(), icon_geom.x(), icon_geom.y(), icon_geom.width() * metric(PdmDevicePixelRatio), icon_geom.height() * metric(PdmDevicePixelRatio),
                               0, attr.depth, InputOutput, visual, mask, &set_attr);
 
 
@@ -196,11 +197,7 @@ void TrayIcon::init()
     XMapWindow(dsp, mIconId);
     XMapRaised(dsp, mWindowId);
 
-    QRect rectangle = iconGeometry();
-    XMoveWindow(QX11Info::display(), mWindowId, rectangle.left(), rectangle.top());
-
     const QSize req_size{mIconSize * metric(PdmDevicePixelRatio)};
-    XResizeWindow(dsp, mWindowId, req_size.width(), req_size.height());
     XResizeWindow(dsp, mIconId, req_size.width(), req_size.height());
 }
 
```